refextract/references/regexs.py

```diff
@@ -648,13 +648,13 @@ def compute_months():
 # This pattern matches both url (http) and 'doi:' or 'DOI' formats
 re_doi = (re.compile(ur"""
     ((\(?[Dd][Oo][Ii](\s)*\)?:?(\s)*)       # 'doi:' or 'doi' or '(doi)' (upper or lower case)
-    |(https?://dx\.doi\.org\/))?            # or 'http://dx.doi.org/'    (neither has to be present)
-    (10\.                                   # 10.                        (mandatory for DOI's)
+    |(https?://(dx\.)?doi\.org\/))?         # or 'http://(dx.)doi.org/'  (neither has to be present)
+    (?P<doi>10\.                            # 10.                        (mandatory for DOI's)
     \d{4}                                   # [0-9] x4
-    /                                       # /
+    (/|%2f)                                 # / (possibly urlencoded)
     [\w\-_:;\(\)/\.<>]+                     # any character
     [\w\-_:;\(\)/<>])                       # any character excluding a full stop
-    """, re.VERBOSE))
+    """, re.VERBOSE + re.IGNORECASE))
```

Comment:
both give the same result if all the flags are different. however it's bad habit that will trip you some day. say there is some code that passes options from user and mixes with your flags and then you have flag1 + flags2 != flags1 | flags2 if they contain the same flag twice. so don't use `+` 
e.g.
`re.X + re.VERBOSE != re.X | re.VERBOSE`

```
In [2]: re.X + re.VERBOSE
Out[2]: 128

In [3]: re.X | re.VERBOSE
Out[3]: 64
```